agent/php_txn.c

```diff
@@ -669,6 +669,62 @@ static void nr_php_txn_send_metrics_once(nrtxn_t* txn TSRMLS_DC) {
 #undef FMT_BOOL
 }
 
+void nr_php_txn_create_agent_version_metric(nrtxn_t* txn, const char* version) {
+  char* metric_name = NULL;
+
+  if (NULL == txn) {
+    return;
+  }
+
+  if (nr_strempty(version)) {
+    return;
+  }
+
+  metric_name = nr_formatf("Supportability/PHP/AgentVersion/%s", version);
+  nrm_force_add(NRTXN(unscoped_metrics), metric_name, 0);
+  nr_free(metric_name);
+}
+
+void nr_php_txn_create_php_version_metric(nrtxn_t* txn, const char* version) {
+  char* metric_name = NULL;
+
+  if (NULL == txn) {
+    return;
+  }
+
+  if (nr_strempty(version)) {
+    return;
+  }
+
+  metric_name = nr_formatf("Supportability/PHP/Version/%s", version);
+  nrm_force_add(NRTXN(unscoped_metrics), metric_name, 0);
+  nr_free(metric_name);
+}
+
+void nr_php_txn_create_agent_php_version_metrics(nrtxn_t* txn) {
+  char* version = NULL;
+
+  if (NULL == txn) {
+    return;
+  }
+
+#ifdef NR_VERSION
+  version = NR_VERSION;
+#else
+  version = "unknown";
+#endif
```

Comment:
When would NR_VERSION not be defined? Why is this conditional construct needed?

Reply:
Fixed in commit [7ea6c93](https://github.com/newrelic/newrelic-php-agent/pull/947/commits/7ea6c93be430eeadd4671cc76923a6b17e78c5f8).